client/src/actions/dataset.js

```diff
@@ -302,18 +302,43 @@ function deletePendingDatasetFailure(id, error) {
     error,
   };
 }
+function deleteFailedDatasetSuccess(id) {
+  return (dispatch) => {
+    dispatch(removeDataset(id));
+  };
+}
+
+function deleteFailedDatasetFailure(id, error) {
+  return {
+    type: constants.DELETE_DATASET_FAILURE,
+    id,
+    error,
+  };
+}
 
 export function deletePendingDataset(id) {
   return (dispatch) => {
     dispatch(deleteDatasetRequest(id));
     api
-      .del(`/api/job_executions/${id}`)
+      .del(`/api/data-source/job-execution/${id}/status/pending`)
       .then(response => response.json())
       .then(() => dispatch(deletePendingDatasetSuccess(id)))
       .catch(error => dispatch(deletePendingDatasetFailure(id, error)));
   };
 }
 
+export function deleteFailedDataset(id) {
+  return (dispatch) => {
+    dispatch(deleteDatasetRequest(id));
+    api
+      .del(`/api/data-source/job-execution/${id}/status/failed`)
+      .then(response => response.json())
+      .then(() => dispatch(deleteFailedDatasetSuccess(id)))
```

Comment:
you can just use `dispatch(removeDataset(id))` here

Reply:
Fixed!